pkg/ebpf/net_events.go

```diff
@@ -233,24 +232,40 @@ func (t *Tracee) processNetEvents(ctx gocontext.Context) {
 			// continue without checking for error, as packetContext will be valid anyway
 			packetContext, networkThread, _ := t.getPcapContextFromTid(netEventMetadata.HostTid)
 
-			if netEventMetadata.NetEventId == NetPacket {
+			if isNetEvent(netEventMetadata.NetEventId) {
 				var netCaptureData bufferdecoder.NetCaptureData
 				err = netDecoder.DecodeNetCaptureData(&netCaptureData)
 				if err != nil {
 					t.handleError(err)
 					continue
 				}
 
+				// handle net event trace
 				ifaceName := t.netInfo.ifaces[int(netCaptureData.ConfigIfaceIndex)].Name
 				ifaceIdx, err := t.getTracedIfaceIdx(ifaceName)
 				if err == nil && ifaceIdx >= 0 {
-					if t.events[netEventMetadata.NetEventId].emit {
-						evt, err := netPacketProtocolHandler(netDecoder, netEventMetadata, networkThread, "net_packet")
-						if err != nil {
-							t.handleError(err)
-							continue
+					// this packet should be traced. i.e. output the event if chosen by the user.
+
+					evt, err := protocolProcessor(networkThread, netEventMetadata, netDecoder)
+					if err != nil {
+						t.handleError(err)
+						continue
+					}
+
+					// derive events chosen by the user
+					derivatives := t.deriveEvent(evt)
+					for _, derivative := range derivatives {
+						// output derived events
+						select {
+						case t.config.ChanEvents <- derivative:
```

Comment:
By deriving the `net_packet` event from both: `dns_request` and `dns_response`, wouldn't you duplicate the `net_packet` event in case user has chosen that event as well ?

```
# ./dist/tracee-ebpf -o format:json -t net=bridge -t e=net_packet,dns_request,dns_response
```

Wouldn't you receive:

net_packet (request)
net_packet (request derived)
dns_request (request event)
net_packet (response)
net_packet (response derived)
dns_response (response event)

Or have you dealt with this somewhere I couldn't see ?

Thanks

Reply:
the reason i needed the derived event in the first place is beacuse we submit the packet only once from the bpf to the user-space.
we use the `set_net_event_id()` to indicate weather this packet is `dns_request`/`dns_response`, or generic `net_packet`. if its dns packet, we derive a generic `net_packet` event, and thus we have only two events for each packet.